File: apps/web/src/lib/types/agent.ts
export type AgentStatus = "active" | "inactive" | "on_leave";
export type CommissionType = "percentage" | "fixed";

export interface Agent {
  id: string;
  tenant_id: string;
  full_name: string;
  email: string;
  phone: string | null;
  photo_key: string | null;
  photo_url: string | null;
  license_id: string | null;
  license_expiry_at: string | null;
  status: AgentStatus;
  default_commission_type: CommissionType;
  default_commission_value: string;
  linked_user_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface AgentCreateRequest {
  full_name: string;
  email: string;
  phone?: string | null;
  license_id?: string | null;
  license_expiry_at?: string | null;
  default_commission_type?: CommissionType;
  default_commission_value?: string | number;
  linked_user_id?: string | null;
}

export interface AgentUpdateRequest {
  full_name?: string | null;
  email?: string | null;
  phone?: string | null;
  license_id?: string | null;
  license_expiry_at?: string | null;
  default_commission_type?: CommissionType | null;
  default_commission_value?: string | number | null;
  linked_user_id?: string | null;
}

export interface AgentListResponse {
  items: Agent[];
  total: number;
}

export interface AgentListParams {
  status?: AgentStatus;
  q?: string;
  skip?: number;
  limit?: number;
}

export interface PhotoUploadResponse {
  photo_key: string;
  url: string;
}

File: apps/api/schemas/agent.py
"""Agent request/response schemas."""

import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, EmailStr

from apps.api.models.enums import AgentStatus, CommissionType


class AgentCreateRequest(BaseModel):
    full_name: str
    email: EmailStr
    phone: str | None = None
    license_id: str | None = None
    license_expiry_at: date | None = None
    default_commission_type: CommissionType = CommissionType.PERCENTAGE
    default_commission_value: Decimal = Decimal("0")
    linked_user_id: uuid.UUID | None = None


class AgentUpdateRequest(BaseModel):
    full_name: str | None = None
    email: EmailStr | None = None
    phone: str | None = None
    license_id: str | None = None
    license_expiry_at: date | None = None
    default_commission_type: CommissionType | None = None
    default_commission_value: Decimal | None = None
    linked_user_id: uuid.UUID | None = None


class AgentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    full_name: str
    email: str
    phone: str | None = None
    photo_key: str | None = None
    photo_url: str | None = None  # assembled by service; not a DB column
    license_id: str | None = None
    license_expiry_at: date | None = None
    status: AgentStatus
    default_commission_type: CommissionType
    default_commission_value: Decimal
    linked_user_id: uuid.UUID | None = None
    created_at: datetime
    updated_at: datetime


class AgentListResponse(BaseModel):
    items: list[AgentResponse]
    total: int


class PhotoUploadResponse(BaseModel):
    photo_key: str
    url: str
